### DISCUSSION TITLE
Gif display over Bluetooth Low Energy

### DISCUSSION THREAD

**Discussion by meganukebmp** [mrcodetastic/ESP32-HUB75-MatrixPanel-DMA#471] [Show and tell]
I made a little project to send gifs to a panel from my phone over BLE. Had to mess around to get the transfer speeds reasonable (I get around 10kB/s currently). The ESP doesnt seem to care much about running the BLE stack in the background, at least with my 64x64 panel size. Received Gifs get saved to the internal flash memory but this can easily be adapted to use SDIO. The reasoning behind using BLE as opposed to just Bluetooth classic with OBEX is to still be useful for various sensors. In this case I have a GPS module attached to it which I can read or subscribe to over BLE.

There is a companion android app that does the image transfer and subscribes to the GPS characteristic as a demo. It would be nice to have it resize gifs for me in the future. 

https://github.com/meganukebmp/ESP32_BLE_Matrixpanel

**Comment by mrcodetastic** [comment]
Very cool!
